src/coreclr/tools/Common/TypeSystem/Ecma/EcmaSignatureParser.cs

```diff
@@ -616,7 +617,7 @@ public MarshalAsDescriptor ParseMarshalAsDescriptor()
                         _reader.ReadSerializedString();
 
                         // Read managed marshaler name
-                        _reader.ReadSerializedString();
+                        managedMarshallerTypeName = _reader.ReadSerializedString();
```

Comment:
I think you should be able to use `_ecmaModule` and `CustomAttributeTypeNameParser.GetTypeByCustomAttributeTypeName` here to resolve the type.